uppsrc/CtrlCore/X11DHCtrl.cpp

```diff
@@ -274,28 +274,29 @@ void DHCtrl::Terminate(void)
 
 	// Unmaps the window
 	MapWindow(false);
-
+	if(utop) {
 	// gathers data from XWindow (needs Input Context...)
-	XWindow *cw = XWindowFromWindow(top->window);
-
-	// Frees input context as needed
-	if(cw->xic)
-	{
-		XDestroyIC(cw->xic);
-		cw->xic = NULL;
+		XWindow *cw = XWindowFromWindow(utop->window);
+	
+		// Frees input context as needed
+		if(cw->xic)
+		{
+			XDestroyIC(cw->xic);
+			cw->xic = NULL;
+		}
+	
+		// Removes XWindow from Upp list
+		RemoveXWindow(utop->window);
+	
+		// Destroys the window
+		// Not to do, it's done destroying the parent window by X11 system
+	//	XDestroyWindow(Xdisplay, top->window);
+	
+		// Destroys created Top struct
+		delete utop;
+		utop = NULL;
```

Comment:
Use build in DeleteTop() instead. It is also setting top flag, which is missing in this implementaiton.

Reply:
Yes. I have DeleteTop() in my other changes. I just left it in the first block of changes and to trace the top to utop renaming and to sync with macos-cocoa renaming.